packages/rol/adapters/thyra/src/function/ROL_ThyraProductME_Constraint_SimOpt.hpp

```diff
@@ -654,54 +768,104 @@ class ThyraProductME_Constraint_SimOpt : public Constraint_SimOpt<Real> {
       this->update_1(u);
     }
 
-    updateValue = false;
+    if (Teuchos::is_null(value_ptr_))
+      value_ptr_ = c.clone();
+    value_ptr_->set(c);
+
+    computeValue = solveConstraint = false;
   }
 
 
   /** \brief Update constraint functions with respect to Sim variable.
                 x is the optimization variable,
-                flag = true if optimization variable is changed,
+                flag = ??,
                 iter is the outer algorithm iterations count.
    */
-  void update_1( const Vector<Real> &u, bool flag = true, int iter = -1 ) {
-    if (flag == true) {
-      updateValue = true;
-      updateJacobian1 = true;
+  void update_1( const Vector<Real> &u, bool /*flag*/ = true, int iter = -1 ) {
+    if(u_hasChanged(u)) {
+      if(verbosityLevel >= Teuchos::VERB_HIGH)
+        *out << "ROL::ThyraProductME_Constraint_SimOpt::update_1, The State Changed" << std::endl;
+      computeValue = computeJacobian1 = true;
+
+      if (Teuchos::is_null(rol_u_ptr))
+        rol_u_ptr = u.clone();
+      rol_u_ptr->set(u);
     }
+
     if(params != Teuchos::null)
       params->set<int>("Optimizer Iteration Number", iter);
   }
 
   /** \brief Update constraint functions with respect to Opt variable.
                 x is the optimization variable,
-                flag = true if optimization variable is changed,
+                flag = ??,
```

Comment:
Does this doxygen need to be updated?

Reply:
Sorry.. at some point I thought that the flag was signalling whether the optimization variable is changed or not. However, it has a more subtle meaning and it does not appear to be consistent.
ROL team is going to completely refactor the udate function, that's why I left the `??` rather then trying to understand this better. Anyway, I'm going to work more on this as we add support for second derivatives.